### ISSUE TITLE
Files not 'pulling' when creating new branch based on remote origin

### ISSUE THREAD

**Issue by brylie** [jacogr/atom-git-control#109]
# User story
As a developer
When I clone a remote branch I expect it to automatically enable remote tracking and pull the remote changes
so that I can start working on the branch as simply as possible

# Observed behavior
When I click on a remote branch that does not exist locally, a new local branch is created, but the local filesystem does not match the remote filesystem. Also, the local branch does not seem to be tracking the remote branch.

# Expected behavior
When clicking a remote branch, a new local branch of the same name would be created. The locak branch will be set to track the remote branch, and all files/changes in the remote branch will be pulled into the newly created local branch.

**Comment by sylvainlap** [comment]
:+1:

**Comment by frese** [comment]
Yes, I have the same problem ! - I'm pretty sure this used to work flawlessly about 6 month ago .. !?

**Comment by brylie** [comment]
Hey all, I added the --track flag in a pull request. Please test and let me know if it works for you.

**Comment by brylie** [comment]
The --track, or -t, flag was the recommended solution as per the following sources:
* http://stackoverflow.com/a/10358772/1191545
* http://stackoverflow.com/a/14436240/1191545

> As a convenience, --track without -b implies branch creation
>
> If no -b option is given, the name of the new branch will be derived from the remote-tracking branch
> https://git-scm.com/docs/git-checkout

So, the basic command for checking out a remote branch and setting up tracking seems to be:

```
git checkout --track remote/branch
```
